### DISCUSSION TITLE
Undefined Method that actually is defined

### DISCUSSION THREAD

**Discussion by brunoaugusto** [babel/babel#13849] [Q&A] [answered]
*Disclaimer: I wasn't very sure if this should go straight to Bug Report*

I have a class to deal with Github Issues Comments with 3 methods: `getIssueCommentReactions()`, `createIssueCommentReaction()` and `deleteIssueCommentReaction()` — verbose, I know. Each one simply performs a Request using a method of another class that's really just a simple wrapper for Fetch API and is used largely across the project without any issues.

On the other side of the table, I have a class with two methods `voteConcession()` and `voteWithdrawal()` to make use of the methods described above, respectively, giving an upvote/downvote or taking it back (after granted, of course). Those are invoked when some `<a>` and the Event bindings are working fine.

This is how `voteConcession()` looks:

```
voteConcession( comment, isUpvote = true ) {

  const API = new GithubReactionsAPI({ owner: 'github_username', repository: 'github_repository' });

  API.createIssueCommentReaction( comment, ( !!isUpvote ? "+1" : "-1" ) ).then( ( response ) => {

    // Renders the visual part of the upvote/downvote, like changing numbers and colours

  }).catch( error => {
    // Show some UI error
  });
}
```

And works perfectly but `voteWithdrawal()` on the other hand, even though it does exactly the same thing (literally) except for using `deleteIssueCommentReaction()` instead, yields me an error:

> Uncaught TypeError: d.deleteIssueCommentReaction(...) is undefined

> I've opened the compiled file and `d` refers to the `const API`

It *could* be just a typo, right? But no. What's puzzling me is that, as I described, the method not only does exist but, on the console, right after the error message, the DELETE Request performed with the Fetch API in this allegedly "undefined method" is executed perfectly (so much that is I open the repository on Github to check, votes are cast/removed)

But, if the method is undefined, how the Request is being performed in the first place? o.O

I've checked as much as my knowledge allowed me: Class/variables names, paths, include order — everything is included **before** the Class with `voteConcession()` and `voteWithdrawal()`, to be sure everything's ready at its disposal — tried to Google the subject as well as on Babel's Issues without any luck, even tried to "debug" the compiled and minified code (not that I could read much of the minified and mangled code) all to no avail.

Also, except for Babel, everything is done the old ways, meaning, there's no React, Vue, Angular, Grunt, Gulp, Webpack and whatever 3rd-party external tool/system you might imagine. Even Node.js is kinda off the table, being used only to run Babel CLI and nothing else.

Thank you for your time

**Comment by JLHwung** [comment, marked as answer]
> Uncaught TypeError: d.deleteIssueCommentReaction(...) is undefined

The type error means the call expression `d.deleteIssueCommentReaction(...)` returns `undefined`, not `d.deleteIssueCommentReaction` is undefined, otherwise the JS will throw `d.deleteIssueCommentReaction` is not a function.

Did you return a Promise in `deleteIssueCommentReaction`?

**Reply by brunoaugusto** [reply]
\*facepalm\*

I... feel like an idiot right now for having tried so many times to understand what was going on and it was a simple `return` statement missing.

Well, living and learning. Thank you very much, have a wonderful weekend :)

**Reply by JLHwung** [reply]
Thanks! I am glad that you sort it out.
